assertk/src/commonMain/kotlin/assertk/assertions/iterable.kt

```diff
@@ -331,3 +331,18 @@ fun <E, T : Iterable<E>> Assert<T>.single(): Assert<E> {
         }
     }
 }
+
+/**
+ * Asserts the iterable contains exactly one element matching the expected element,
+ * and returns an assert on that element.
+ */
+inline fun <E, T : Iterable<E>> Assert<T>.single(predicate: (E) -> Boolean): Assert<E> {
```

Comment:
No real advantage to this for tests
```suggestion
fun <E, T : Iterable<E>> Assert<T>.single(predicate: (E) -> Boolean): Assert<E> {
```